docs/modules/sql/pages/querying-imap.adoc

```diff
@@ -68,16 +64,78 @@ the following fields:
 |`INTEGER`
 |===
 
-Together with the key and value objects, you may query the following fields from
-`IMap<Long, Employee>`:
+If you create a map whose values are `Employee` objects, you can query the following fields from it:
 
 [source,sql]
 ----
 SELECT __key, this, name, age FROM employee
 ----
 
-If both the key and value have fields with the same name, then the field of the
-value is exposed.
+NOTE: If both the key and the value share a field with the same name, you may only query the value's field and not the key's field.
+
+=== Using Parameterized Queries
+
+A parameterized query is a type of SQL query that requires at least one parameter for execution. Instead of putting data straight into an SQL statement, parameterized queries allow you to use the `?` placeholder. This type of query is often used to mitigate SQL injection attacks. For a discussion on this topic, see this answer on link:https://stackoverflow.com/a/33033576[StackOverflow^].
+
+For example, consider the following example of an `Employee` object.
+
+[cols="1,1"]
+|===
+| Name
+| Type
+
+|`name`
+|`VARCHAR`
+
+|`age`
+|`INTEGER`
+
+|`paycheckInfo`
+|`PayCheckInfo`
+|===
+
+You could create a parameterized query that finds employees who are older than a given integer:
+
+[source,java]
+----
+int ageToCompare = 30;
+sql.execute("SELECT name FROM employee WHERE employee.age > ?", ageToCompare);
+----
+
+In this `Employee` object, the last value is an instance of a `payCheckInfo` class. Here is the structure of that class:
+
+[cols="1,1"]
+|===
+| Name
+| Type
+
+|`startDate`
+|`INTEGER`
+
+|`amount`
+|`INTEGER`
+|===
+
+Whether you can query nested fields in this object depends on how your map entries are serialized:
+
+- For Java objects (xref:serialization:implementing-java-serializable.adoc[`Serializable`], xref:serialization:implementing-dataserializable.adoc[`DataSerializable`], xref:serialization:implementing-java-serializable.adoc#identifieddataserializable[`IdentifiedDataSerializable`]), you must xref:clusters:deploying-code-on-member.adoc[add the object's class to the classpath of the member] you are sending the query to.
+
+- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, you do not need to add the object's class to the classpath of the member, unless you want to use parameterized queries to compare one object against another.
+
+For example, with `Portable` objects, you could run the following queries without adding the `payCheckInfo` class to the member's classpath:
+[source,sql]
+----
+SELECT paycheckInfo FROM employee WHERE employee.age > 10
+SELECT paycheckInfo FROM employee WHERE employee.paycheckInfo.amount > 1000

```

Comment:
We don't support nested fields (yet). The second query won't work.

Reply:
@gierlachg is this still true for 5.0? And does this apply to all serialization methods?

Reply:
Yes, nested fields are not supported yet.